spec/controllers/users/registrations_controller_spec.rb

```diff
@@ -0,0 +1,250 @@
+      end
+
+      it 'offers to manage OmniAuth accounts' do
+        get :edit
+        expect(response.body).to include(I18n.t('users.registrations.edit.manage_omniauth', kind: OmniAuth::Utils.camelize(provider)))
+      end
+
+      context 'when no identity is linked' do
+        it 'does not offer to unlink the identity' do
+          get :edit
+          expect(response.body).not_to include(I18n.t('users.registrations.edit.remove_identity', kind: OmniAuth::Utils.camelize(provider)))
+        end
+
+        it 'offers to add the identity' do
+          get :edit
+          expect(response.body).to include(I18n.t('users.registrations.edit.add_identity', kind: OmniAuth::Utils.camelize(provider)))
+        end
+
+        it 'does not explain that the last identity cannot be removed' do
+          get :edit
+          expect(response.body).not_to include(I18n.t('users.registrations.edit.cannot_remove_last_identity', kind: OmniAuth::Utils.camelize(provider)))
+        end
+      end
+
+      context 'when an existing identity is linked' do
+        before { create(:user_identity, user:, omniauth_provider: provider) }
+
+        it 'offers to unlink the identity' do
+          get :edit
+          expect(response.body).to include(I18n.t('users.registrations.edit.remove_identity', kind: OmniAuth::Utils.camelize(provider)))
+        end
+
+        it 'does not offer to add the identity' do
+          get :edit
+          expect(response.body).not_to include(I18n.t('users.registrations.edit.add_identity', kind: OmniAuth::Utils.camelize(provider)))
+        end
+
+        it 'does not explain that the last identity cannot be removed' do
+          get :edit
+          expect(response.body).not_to include(I18n.t('users.registrations.edit.cannot_remove_last_identity', kind: OmniAuth::Utils.camelize(provider)))
+        end
+
+        context 'when no password is set' do
+          before { user.update(password_set: false) }
+
+          it 'does not offer to unlink the identity' do
+            get :edit
+            expect(response.body).not_to include(I18n.t('users.registrations.edit.remove_identity', kind: OmniAuth::Utils.camelize(provider)))
+          end
+
+          it 'does not offer to add the identity' do
+            get :edit
+            expect(response.body).not_to include(I18n.t('users.registrations.edit.add_identity', kind: OmniAuth::Utils.camelize(provider)))
+          end
+
+          it 'explains that the last identity cannot be removed' do
+            get :edit
+            expect(response.body).to include(I18n.t('users.registrations.edit.cannot_remove_last_identity', kind: OmniAuth::Utils.camelize(provider)))
+          end
+        end
+      end
+    end
+  end
+
+  describe '#update' do
+    context 'when avatar was removed' do
+      before { user.avatar.attach(io: Rails.root.join('spec/fixtures/files/red.bmp').open, filename: 'red.bmp') }
```

Comment:
If we merge #1317 first, I can change this path to one of the avatars uploaded there.

Reply:
I made the required change, so that we are using a real avatar here.

Reply:
I forgot to rebase after the OmniAuth specs were merged (since I did the changes twice, locally). Fixed that now, new test pipeline is running.